### PR TITLE
Precise memory & Exit on time

### PR DESCRIPTION
**Pull request by chenyaojie**
1. Make memory usage more precise;
2. Allows the program to terminate on time.

### CODE REVIEW

File: 13/memory-user.c [hunk 1]
```diff
@@ -3,26 +3,33 @@
...
     clock_t begin = clock();
     double time_spent;
 
-    for (int i = 0; i < length; i++) {
+    for (int i = 0; i < length; i++)
+    {
+        arr[i] += 1;
+    }
+
+    while (1)
```
Comment (xxyzz) [inline 13/memory-user.c:28]: Move `malloc`, `for` and `free`  inside `while` is more reasonable, IMHO.

Comment (chenyaojie) [reply 13/memory-user.c:28]: Awesome i agree

**Comment by xxyzz** [comment]
Thanks for your contribution!

**Comment by xxyzz** [comment]
If the `for` loop executes longer than the time specified, this implication is still inaccurate.

**Comment by chenyaojie** [comment]
> Thanks for your contribution!

I feel very happy to discuss with you 😀

**Pull request merged**